src/charm.py

```diff
@@ -278,6 +333,21 @@ def _on_get_rule_action(self, event: ActionEvent) -> None:
         event.log(f"Successfully fetched rule: {rule_id}")
         event.set_results(rule)
 
+    def _on_invalid_forward_auth_config(self, event: InvalidForwardAuthConfigEvent) -> None:
+        logger.info(
+            "The forward-auth config is invalid: one or more of the related apps is missing ingress relation"
+        )
+        self.unit.status = BlockedStatus(event.error)
+
+    def _on_forward_auth_proxy_set(self, event: ForwardAuthProxySet) -> None:
+        logger.info("The proxy was set successfully")
+        self.unit.status = ActiveStatus("Identity and Access Proxy is ready")
+
+    def _on_forward_auth_relation_removed(self, event: ForwardAuthRelationRemovedEvent) -> None:
+        logger.info("The proxy was unset")
+        # The proxy was removed, but the charm is still functional
+        self.unit.status = ActiveStatus()
```

Comment:
Should oathkeeper be in waiting state until the `ForwardAuthProxySet` event is emitted (which means that both the access rules and traefik routes are in place) and when the `forward-auth` relation has departed?
When neither of the rules are created, Oathkeeper is still functional, but it does not provide the IAP features, so I'm hesitant whether `ActiveStatus` sounds correct.